src/js/components/Header.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { Link, HashRouter } from 'react-router-dom';
+
+class Header extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      authenticated: false
+    };
+    this.logout = this.logout.bind(this);
```

Comment:
no need to bind `this` within the constructor, es6 allows you to use arrow function when defining the method. Binding `this` allows you to keep context of what `this` you are referring to. With the advent of arrow functions, you can define functions without losing context of `this`. You can remove this line and modify your logout definition as shown in the next comment below...